### DISCUSSION TITLE
Git secret is not propagated to ArgoCD

### DISCUSSION THREAD

**Discussion by acanizares** [akuity/kargo#2977] [Q&A] [answered]
I'm following [Kargo's quickstart](https://docs.kargo.io/quickstart/) with a Kind cluster, and the first promotion gets stuck in the `argocd-update` step.

When I inspect the ArgoCD dashboard I see the error
```
Failed to load target state: failed to generate manifest for source 1 of 1: rpc error: code = Unknown desc = authentication required
```

If I then provide the git credentials directly to ArgoCD (via `Settings -> Repositories -> Connect repo`) the application sync succeeds. But I think ArgoCD should be using the git secret directly?

**Comment by krancour** [comment, marked as answer]
They are not meant to be shared in that way.

You must have made your fork of the kargo-demo-gitops private? That wasn't the intention, and the quickstart may need an update to clarify this point.

Assuming it had been public, Kargo needed credentials to _write_ to the repo, but Argo CD would have required none to _read_ from it. And this actually illustrates one of the underlying reasons that the two systems do not share secrets. Argo CD only ever needs read permissions and Kargo frequently needs write permissions. By the principle of least privilege, they ought not share or else you've given Argo CD more permissions than it requires.

**Reply by acanizares** [reply]
I see! I was using a private repository. Thanks for clarifying 🙏
